component/src/main/java/org/wso2/carbon/identity/authenticator/pinterest/PinterestAuthenticator.java

```diff
@@ -0,0 +1,396 @@
+		}
+		StringBuilder stringBuilder = new StringBuilder();
+		BufferedReader bufferedReader = null;
+		HttpURLConnection httpConnection = null;
+		try {
+			URL obj = new URL(url + PinterestAuthenticatorConstants.QUESTION_MARK +
+			                  PinterestAuthenticatorConstants.ACCESS_TOKEN + PinterestAuthenticatorConstants.EQUAL +
+			                  accessToken);
+			URLConnection connection = obj.openConnection();
+			// Cast to a HttpURLConnection
+			if (connection instanceof HttpURLConnection) {
+				httpConnection = (HttpURLConnection) connection;
+				httpConnection.setRequestMethod(PinterestAuthenticatorConstants.HTTP_GET_METHOD);
+				bufferedReader = new BufferedReader(new InputStreamReader(httpConnection.getInputStream()));
+			} else {
+				throw new ApplicationAuthenticatorException("Exception while casting the HttpURLConnection");
+			}
+			String inputLine = bufferedReader.readLine();
+			while (inputLine != null) {
+				stringBuilder.append(inputLine).append("\n");
+				inputLine = bufferedReader.readLine();
+			}
+		} catch (MalformedURLException e) {
+			throw new ApplicationAuthenticatorException("Exception while generating the user info URL", e);
+		} catch (ProtocolException e) {
+			throw new ApplicationAuthenticatorException("Exception while setting the request method", e);
+		} catch (IOException e) {
+			throw new ApplicationAuthenticatorException("Exception while reading the response", e);
+		} finally {
+			IdentityIOStreamUtils.closeReader(bufferedReader);
+			if (httpConnection != null) {
+				httpConnection.disconnect();
+			}
+		}
+		if (log.isDebugEnabled()) {
+			log.debug("Receiving the response for the User info: " + stringBuilder.toString());
+		}
+		return stringBuilder.toString();
+	}
+
+	/**
+	 * Get the Pinterest specific claim dialect URI.
+	 *
+	 * @return Claim dialect URI.
+	 */
+	@Override
+	public String getClaimDialectURI() {
+		return PinterestAuthenticatorConstants.CLAIM_DIALECT_URI;
+	}
+
+	/**
+	 * @param token                   token
+	 * @param authenticatorProperties authenticatorProperties
+	 * @return claims
+	 */
+	private Map<ClaimMapping, String> buildClaims(OAuthClientResponse token,
+	                                              Map<String, String> authenticatorProperties)
+			throws ApplicationAuthenticatorException {
+		Map<ClaimMapping, String> claims = new HashMap<>();
+		String accessToken = token.getParam("access_token");
+		String url = getUserInfoEndpoint(token, authenticatorProperties);
+		String json;
+		try {
+			json = getUserInfo(url, accessToken);
+
+			if (StringUtils.isBlank(json)) {
+				if (log.isDebugEnabled()) {
+					log.debug("Unable to fetch user claims. Proceeding without user claims");
+				}
+				return claims;
+			}
+			JSONObject obj = new JSONObject(json);
+			String userData = obj.getJSONObject(PinterestAuthenticatorConstants.ROOT_ELEMENT).toString();
+			Map<String, Object> jsonObject = JSONUtils.parseJSON(userData);
```

Comment:
Why 
String userData = json.toString()
and then
JSONUtils.parseJSON(userData);
This heavy on performance (CPU and memory)

We should be able to traverse the JSON DOM itself the get them.